### ISSUE TITLE
400 Bad Request The redirect URI you provided has not been whitelisted for your application. Please add your redirect URI in the 'Allowed Return URLs' section under 'Web Settings' for your Security Profile on Amazon Developer Portal.

### ISSUE THREAD

**Issue by kevinkey619b** [alexa/alexa-avs-sample-app#744]
Ugh,

Kept having authentication issues, so I deleted everything and started over from the beginning. Now it seems that Amazon is block all my authentications. Here's the error message. Is there any workaround?

400 Bad Request
The redirect URI you provided has not been whitelisted for your application. Please add your redirect URI in the 'Allowed Return URLs' section under 'Web Settings' for your Security Profile on Amazon Developer Portal.

**Comment by kevinkey619b** [comment]
So I have tried just about every Linux-compatible browser I can find and I still can't get this thing to work. I don't know what is wrong. Maybe I'm banned by Amazon?

**Comment by Kinjalk** [comment]
Can you provide screenshot of your Amazon Developer Portal where you have set URLs ?

**Comment by kevinkey619b** [comment]
Annnndddd... many hours later... I finally got it all working by creating an entire new device in Amazon and naming it "my_device". I don't know what else was going wrong! I made sure to not confuse Client ID and Client Secret, made sure to add the Allowed Origin of https://localhost:3000 and Return URL of https://localhost:3000/authresponse, and more. After many many many complete deletions of the install folder on my RPi, many reboots, trying various Linux web browsers, etc.  it seems creating an entire new device in https://developer.amazon.com is what finally got things working again.
